mdnsbridge/mdnsbridgeclient.py

```diff
@@ -76,28 +76,50 @@ def getHref(self, srv_type, priority=None, api_ver=None, api_proto=None):
                 continue
             if api_proto is not None and api_proto != service["protocol"]:
                 continue
+
             if priority >= 100:
                 if service["priority"] == priority:
-                    return self._createHref(service)
+                    return [service]
             else:
                 if service["priority"] < current_priority:
                     current_priority = service["priority"]
                     valid_services = []
                 if service["priority"] == current_priority:
                     valid_services.append(service)
+
         if len(valid_services) == 0:
             self.logger.writeWarning("No services found: {}".format(srv_type))
             if srv_type == "nmos-query":
                 return self._checkLocalQueryServiceExists()
             return ""
 
-        # Randomise selection. Delete entry from the services list and return it
+        # Randomise selection.
         random.seed()
-        index = random.randint(0, len(valid_services)-1)
-        service = valid_services[index]
-        href = self._createHref(service)
-        self.services[srv_type].remove(service)
-        return href
+        random.shuffle(valid_services)
+        return valid_services
+
+    def getHrefList(self, srv_type, priority=None, api_ver=None, api_proto=None):
+        """Get a list of valid service hrefs for the priority, service type and version specified"""
+        valid_services = self._getServiceList(srv_type, priority, api_ver, api_proto)
+
+        if type(valid_services) is list:
+            href_list = []
+            for service in valid_services:
+                href_list.append(self._createHref(service))
+                self.services[srv_type].remove(service)
```

Comment:
Is there going to be an issue here if an application (for some reason) called both `getHrefList` and `getHref`? It looks like `getHref` may gradually empty `self.services` so that `getHrefList` isn't guaranteed to return the full set. I wonder if `_getServiceList` should have an option to force a call to `_updateServices`, or perhaps the check for whether `self.services` contains any valid entries should only exist within `getHref` rather than inside `_getServiceList`?